src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -493,4 +511,56 @@ public static Object coerceBodyType(InputStream body, ResponseBodyType as,
 
         return response;
     }
+
+    private static Timer.Context startTimer(MetricRegistry registry, HttpRequest request) {
+        if (registry != null) {
+            final RequestLine requestLine = request.getRequestLine();
+            final String name = MetricRegistry.name("puppetlabs.http-client", requestLine.getUri(),
+                    requestLine.getMethod());
```

Comment:
Hmm. What else are you thinking about having under the URI?

I was thinking that if you have a URI that can take multiple methods, you probably care more about grouping based on URI and then based on method than based on method first.

I think that having multiple timers going down the hierarchy probably makes sense. I guess one thought to counter the previous statement I made about request method is that if you had the ability to do both GET and POST at like /users/<user id>, you might want to get information about all GET/POST requests to just `/users/*`, and that would be easiest if we had timers along the hierarchy for `GET./users./<user-id>` and `POST./users./<user-id>` than for something like `/users./<user-id>.GET`.